hs_core/tasks.py

```diff
@@ -2,45 +2,49 @@
 
 from __future__ import absolute_import
 
+import logging
 import os
 import sys
 import traceback
 import zipfile
-import logging
-
-import requests
-
+from datetime import timedelta, date
 from xml.etree import ElementTree
 
-from rest_framework import status
-
+import requests
+from celery import shared_task
+from celery.schedules import crontab
+from celery.task import periodic_task
 from django.conf import settings
-from django.core.mail import send_mail
 from django.contrib.auth.models import User
+from django.core.mail import send_mail
+from rest_framework import status
 
-from celery.task import periodic_task
-from celery.schedules import crontab
-from celery import shared_task
-
-from hs_core.models import BaseResource
+from django_irods.icommands import SessionException
+from django_irods.storage import IrodsStorage
 from hs_core.hydroshare import utils
 from hs_core.hydroshare.hs_bagit import create_bag_files
 from hs_core.hydroshare.resource import get_activated_doi, get_resource_doi, \
     get_crossref_url, deposit_res_metadata_with_crossref
-from django_irods.storage import IrodsStorage
+from hs_core.models import BaseResource
 from theme.models import UserQuota, QuotaMessage
 from theme.utils import get_quota_message
 
-from django_irods.icommands import SessionException
-
-
 # Pass 'django' into getLogger instead of __name__
 # for celery tasks (as this seems to be the
 # only way to successfully log in code executed
 # by celery, despite our catch-all handler).
 logger = logging.getLogger('django')
 
 
+@periodic_task(ignore_result=True, run_every=crontab(minute=0, hour=0))
+def nightly_zips_cleanup():
+    # delete 2 days ago to avoid deleting zips just before midnight
+    date_folder = (date.today() - timedelta(2)).strftime('%Y-%m-%d')
+    irods_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    zips_daily_date = "{irods}/zips/{daily_date}".format(irods=irods_prefix, daily_date=date_folder)
+    IrodsStorage().delete(zips_daily_date)

```

Comment:
@sblack-usu The ```delete``` method does not check whether the folder exists or not, so if it does not exist, it will raise an iRODS exception. To avoid this, you can call exists() method to only delete when this folder does exist.

Reply:
@sblack-usu I am wondering whether we should move this periodic task into ```manage_task_nightly()``` since they are scheduled to run at exact same time. I am thinking this may have performance benefit since only one task needs to be spawned to handle these different nightly run operations.

Reply:
I thought about that too, but I felt the code was cleaner having it in it's own task.  We also run the risk of jobs clustered in one task risk not being run if an upstream job has an unhandled exception.  If efficiency is a problem/concern, we could always just run it at a different time.  This task should run very quickly, it just deletes one folder.

Reply:
@sblack-usu I see that also. I vote for running it at a different time just in case, perhaps 30 minutes apart.

Reply:
I updated it to run at 23:30